liver_imaging_analysis/engine/experiment_tracking.py

```diff
@@ -0,0 +1,327 @@
+import os
+import sys
+import shutil
+from googleapiclient.discovery import build
+from googleapiclient.http import MediaFileUpload
+from googleapiclient.errors import HttpError
+from google.oauth2.credentials import Credentials
+from google_auth_oauthlib.flow import InstalledAppFlow
+from google.auth.transport.requests import Request
+from torch.utils.tensorboard import SummaryWriter
+from clearml import Task, Dataset
+from liver_imaging_analysis.engine.config import config
+
+# If modifying these scopes, delete the file token.json.
+SCOPES = ["https://www.googleapis.com/auth/drive.file"]
+
+
+class ExperimentTracking:
+    """
+    Class for experiment tracking using ClearML, Tensorboard and Google Drive.
+
+    Args:
+        experiment_name (str): Name of the experiment.
+        run_name (str): Name of the run within the experiment.
+    """
+
+    def __init__(self, hparams):
+        """
+        Initializes an ExperimentTracking instance for the chosen model and hyperparameters.
+
+        Parameters
+        ----------
+        hparams : dict
+            A dictionary containing the hyperparameters used for the experiment.
+        """
+        # clearml credentials for experiment tracking
+        Task.set_credentials(
+            api_host="https://api.clear.ml",
+            web_host="https://app.clear.ml",
+            files_host="https://files.clear.ml",
+            key= config.clearml_credentials["key"],
+            secret= config.clearml_credentials["secret"],
+        )
+        self.experiment_name, self.run_name = self.exp_naming(hparams)
+        self.task=None
+        self.writer=None
+        self.hparams=hparams
+    
+    def Hash(self, text:str):
```

Comment:
use appropriate naming conventions ...

Reply:
changed hparams to hyperparameters

Reply:
I actually meant the name of the function Hash() should start with small letter

Reply:
Ok, done